doc/xbgpu.design.b.rst

```diff
@@ -86,3 +86,35 @@ scaling to convert to a real value in :math:`(-0.5, 0.5)`. While this is
 still a deviation from uniformity, it does give a symmetric distribution.
 
 .. _curand: https://docs.nvidia.com/cuda/curand/index.html
+
+Data flow
+---------
+The host side of the beamforming is simpler than for correlation, because
+there is no accumulation. For simplicity, the output chunk size (in time) is
+set to the same as the input chunk size.
+
+Because the kernel operates on all beams together, there is only one instead
+of the :class:`.BPipeline` class, and it handles all the beams. This is in
```

Comment:
Argh, am I working on an outdated old English-teacher's tale of "Never start a sentence with *Because*" ? May I suggest something like
> The beamforming kernel operates on all beams together. As a result, there is only one instance of the :class:`.BPipeline` class, which also handles all the beam :class:`Output`s.

I mention :class:`Output`s at the end because it ties in more tangibly with the ending of the next sentence.

Reply:
Bruce's sentence is grammatically correct. I don't have my copy of Fowler with me (at the office at the moment) but this is simply a rearrangement of the sentence "There is only one instance of the Bpipeline class because the kernel operates on all the beams together", you're just emphasising the "the kernel operates on all the beams together" part.